app/lib/features/pins/pages/create_pin_page.dart

```diff
@@ -0,0 +1,353 @@
+          border: Border.all(
+            color: Theme.of(context).colorScheme.primary,
+            strokeAlign: BorderSide.strokeAlignOutside,
+          ),
+          borderRadius: BorderRadius.circular(12),
+        ),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            Icon(
+              Atlas.file_arrow_up_thin,
+              size: 14,
+              color: Theme.of(context).colorScheme.neutral5,
+            ),
+            Text(
+              'Upload File',
+              style: Theme.of(context)
+                  .textTheme
+                  .labelLarge!
+                  .copyWith(color: Theme.of(context).colorScheme.neutral5),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _buildLinkField() {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: <Widget>[
+        const Padding(
+          padding: EdgeInsets.only(bottom: 5),
+          child: Text('Link'),
+        ),
+        InputTextField(
+          hintText: 'https://',
+          key: CreatePinPage.urlFieldKey,
+          textInputType: TextInputType.url,
+          controller: _linkController,
+          validator: (value) =>
+              hasLinkOrText() ? null : 'Text or URL must be given',
+        ),
+      ],
+    );
+  }
+
+  Widget _buildDescriptionField() {
+    final labFeature = ref.watch(featuresProvider);
+    bool isActive(f) => labFeature.isActive(f);
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: <Widget>[
+        const Padding(
+          padding: EdgeInsets.only(bottom: 5),
+          child: Text('Description'),
+        ),
+        Visibility(
+          visible: isActive(LabsFeature.pinsEditor),
+          replacement: SizedBox(
+            height: 200,
+            child: MdEditorWithPreview(
+              key: CreatePinPage.contentFieldKey,
+              validator: (value) =>
+                  hasLinkOrText() ? null : 'Text or URL must be given',
+              controller: _textController,
+            ),
+          ),
+          child: Container(
+            height: 200,
+            margin: const EdgeInsets.symmetric(vertical: 8),
+            decoration: BoxDecoration(
+              color: Theme.of(context).colorScheme.primaryContainer,
+              borderRadius: BorderRadius.circular(12),
+            ),
+            constraints:
+                BoxConstraints(maxHeight: MediaQuery.of(context).size.height),
+            child: HtmlEditor(
+              editable: true,
+              autoFocus: false,
+              shrinkWrap: true,
+              editorState: EditorState.blank(),
+              footer: const SizedBox(),
+              onChanged: (body, html) {
+                if (html != null) {
+                  _textController.text = html;
+                }
+                _textController.text = body;
```

Comment:
this is what I mean. `html` here is explicit and rather than keeping that explicit data available, it and the body is just crammed into the same `_textController.text` field.

Also, this is _never_ holding the `html` as the second `_textController.text` is always overwriting it.

Instead I'd recommend have separate _textController for each:

```suggestion
                if (html != null) {
                  _htmlTextController.text = html;
                } else {
                 _htmlTextController.text = null;
                }
                _textController.text = body;
```